lib/ansible/modules/crypto/entrust/ecs_certificate.py

```diff
@@ -0,0 +1,906 @@
+              be renewed or reissued and saved to I(path)
+            - If there is not a certificate present in I(path) and the I(force) and I(remaining_days) parameters do not indicate a new certificate is needed,
+              the certificate referenced by I(tracking_id) certificate will be saved to I(path).
+            - This can be used when a known certificate is not currently present on a server, but you want to renew or reissue it to be managed by an ansible
+              playbook. For example, if you specify C(request_type=RENEW), I(tracking_id) of an issued certificate, and I(path) to a file that does not exist,
+              the first run of a task will download the certificate specified by I(tracking_id) (assuming it is still valid), and future runs of the task will
+              (if applicable - see I(force) and I(remaining_days)) renew the certificate now present in I(path).
+        type: int
+    remaining_days:
+        description:
+            - The number of days the certificate must have left being valid. If C(cert_days < remaining_days) then a new certificate will be
+              obtained using I(request_type).
+            - If C(request_type=RENEW), a renew will fail if the certificate being renewed has been issued within the past 30 days, so do not set a
+              I(remaining_days) value that is within 30 days of the full lifetime of the certificate being acted upon. (e.g. if you are requesting Certificates
+              with a 90 day lifetime, do not set remaining_days to a value C(60) or higher).
+            - The I(force) option may be used to ensure that a new certificate is always obtained.
+        type: int
+        default: 30
+    request_type:
+        description:
+            - Operation performed if I(tracking_id) references a valid certificate to reissue, or there is already a certificate present in I(path) but either
+              I(force) is specified or C(cert_days < remaining_days).
+            - Specifying C(request_type=NEW) means a certificate request will always be submitted and a new certificate issued.
+            - Specifying C(request_type=RENEW) means that an existing certificate (specified by I(tracking_id) if present, otherwise I(path)) will be renewed.
+              If there is no certificate to renew, a new certificate is requested.
+            - Specifying C(request_type=REISSUE) means that an existing certificate (specified by I(tracking_id) if present, otherwise I(path)) will be
+              reissued.
+              If there is no certificate to reissue, a new certificate is requested.
+            - If a certificate was issued within the past 30 days, the 'renew' operation is not a valid operation and will fail.
+            - Note that C(reissue) is an operation that will result in the revocation of the certificate that is reissued, be cautious with it's use.
+            - I(check_mode) is only supported if C(request_type=NEW)
```

Comment:
What happens in check mode if `request_type` is not `NEW`?

Reply:
In any case, the module should *not* actually do something when check_mode is `True`.